### PR TITLE
Partial deflaking of abstract_method_smoke_test

### PR DESCRIPTION
**Pull request by mklim**
## Description

1. The platform code to show the keyboard wasn't working on certain
devices. From my testing it appears to be related to when the code was
firing. IMM won't show the soft input (or shows and then immediately
hides it, it's hard to tell) if it's called before the Flutter UI is
loaded. Change this to instead show the soft keyboard after a message
from Flutter that main() has been started.

2. A text field was visible in the UI, and the test was run under a
fuzzer that random tapped portions of the screen. Remove the text field
so that the fuzzer can't accidentally open the keyboard on its own at a
random time.

3. The keyboard was left open even after the app was closed. Also toggle
the input off when the app was closed, so that this test can be ran
multiple times in succession relatively hermetically.

## Related Issues

#42349

## Tests

I added the following tests:

- Updated abstract_method_smoke_test

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require Flutter developers to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (Please read [Handling breaking changes]). *Replace this with a link to the e-mail where you asked for input on this proposed change.*
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

**Review by blasten — APPROVED** [review]
Thanks!

File: dev/integration_tests/abstract_method_smoke_test/android/app/src/main/kotlin/com/example/abstract_method_smoke_test/MainActivity.kt [hunk 1]
```diff
@@ -12,8 +14,22 @@ class MainActivity: FlutterActivity() {
     GeneratedPluginRegistrant.registerWith(this)
 
     // Triggers the Android keyboard, which causes the resize of the Flutter view.
-    // https://github.com/flutter/flutter/issues/40126
+    // We need to wait for the app to complete.
+    MethodChannel(getFlutterView(), "com.example.abstract_method_smoke_test")
```
Comment (dnfield) [inline dev/integration_tests/abstract_method_smoke_test/android/app/src/main/kotlin/com/example/abstract_method_smoke_test/MainActivity.kt]: Is this a bug?  Why can't an app show the keyboard before Flutter is ready?

File: dev/integration_tests/abstract_method_smoke_test/lib/main.dart [hunk 2]
```diff
@@ -48,9 +52,8 @@ class SecondPage extends StatelessWidget {
   Widget build(BuildContext context) {
     return Scaffold(
       body: Column(
-        children: <Widget>[
-          TextFormField(),
-          const Expanded(
+        children: const <Widget>[
+          Expanded(
```
Comment (dnfield) [inline dev/integration_tests/abstract_method_smoke_test/lib/main.dart]: This seems like a problem with the embedding - tapping on a text field at the wrong moment shouldn't cause a WAKE_LOCK or a crash to fire, right?

File: dev/integration_tests/abstract_method_smoke_test/android/app/src/main/kotlin/com/example/abstract_method_smoke_test/MainActivity.kt (hunk 1 above)
Comment (dnfield) [reply dev/integration_tests/abstract_method_smoke_test/android/app/src/main/kotlin/com/example/abstract_method_smoke_test/MainActivity.kt]: Perhaps the keyboard is getting hidden from the fuzzer tapping randomly (and maybe tapping on the text input and then away from it...)

**Review by dnfield — APPROVED** [review]

**Pull request merged**